File: CycleLength/solve/ruby/CycleLengthSolver.rb
#! /usr/bin/env ruby

require_relative "CycleLength"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  seed = reader.next(TInt)
  reader.next()
  a = reader.next(TInt)
  reader.next()
  b = reader.next(TInt)
  reader.next()
  m = reader.next(TInt)

  result = CycleLength.new().solve(seed, a, b, m)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main

File: CycleLength/solve/python/CycleLengthSolver.py
#! /usr/local/bin/python3

import os, sys
import CycleLength

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        seed = reader.next("int")
        reader.next()
        a = reader.next("int")
        reader.next()
        b = reader.next("int")
        reader.next()
        m = reader.next("int")

    result = CycleLength.solve(seed, a, b, m)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "int"))

if __name__ == "__main__":
    init()
    main()
